### PR TITLE
Remove `--update-proposal-file` from Conway onwards

### PR DESCRIPTION
**Pull request by newhoggy**
# Changelog

```yaml
- description: |
    Remove `--update-proposal-file` from Conway onwards
# uncomment types applicable to the change:
  type:
  # - feature        # introduces a new feature
  # - breaking       # the API has changed in a breaking way
  - compatible     # the API has changed but is non-breaking
  # - optimisation   # measurable performance improvements
  - improvement    # QoL changes e.g. refactoring
  # - bugfix         # fixes a defect
  # - test           # fixes/modifies tests
  # - maintenance    # not directly related to the code
  # - release        # related to a new release preparation
  # - documentation  # change in code docs, haddocks...
```

# Context

This option is not supported from Conway onwards to remove it from the CLI parser rather than accept it and then fail validation.

# How to trust this PR

Highlight important bits of the PR that will make the review faster. If there are commands the reviewer can run to observe the new behavior, describe them.

# Checklist

- [ ] Commit sequence broadly makes sense and commits have useful messages
- [ ] New tests are added if needed and existing tests are updated. See [Running tests](https://github.com/input-output-hk/cardano-node-wiki/wiki/Running-tests) for more details
- [ ] Self-reviewed the diff

### CODE REVIEW

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs [hunk 1]
```diff
@@ -3246,6 +3246,18 @@ pTxId l h =
 -- Helpers
 --------------------------------------------------------------------------------
 
+pFeatured :: ()
+  => Eon eon
+  => ToCardanoEra peon
+  => peon era
+  -> Parser a
+  -> Parser (Maybe (Featured eon era a))
```
Comment (carlhammann) [inline cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs:3254]: I think this is a good trick. With the context of this PR it is clear what this function accomplishes, but I feel a comment to the effect that this parser _doesn't show up_ in the command line if the era is not in the eon is warranted.

File: cardano-cli/src/Cardano/CLI/Types/Errors/TxValidationError.hs [hunk 2]
```diff
@@ -342,6 +335,16 @@ validateTxScriptValidity era (Just scriptValidity) = do
...
 
+-- TODO legacy. This can be deleted when legacy commands are removed.
+validateUpdateProposalFile
+  :: CardanoEra era
+  -> Maybe UpdateProposalFile
+  -> Either TxUpdateProposalValidationError (Maybe (Featured ShelleyToBabbageEra era (Maybe UpdateProposalFile)))
+validateUpdateProposalFile era = \case
+  Nothing -> pure Nothing
+  Just updateProposal -> do
+    supported <- conjureWitness era TxUpdateProposalNotSupported
```
Comment (carlhammann) [inline cardano-cli/src/Cardano/CLI/Types/Errors/TxValidationError.hs:346]: Ah, there it is. I was wondering why `TxUpdateProposalNotSupported` wasn't deleted. :smile:

**Review by carlhammann — APPROVED** [review]
LGTM

File: cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs (hunk 1 above)
Comment (newhoggy) [reply cardano-cli/src/Cardano/CLI/EraBased/Options/Common.hs:3254]: I think documentation is good, but I didn't want to repeat myself by for example writing the same comment for every such parser we define.  Is there something else we can do?

**Review by carbolymer — APPROVED** [review]

**Review by CarlosLopezDeLara — APPROVED** [review]

**Review by Jimbo4350 — APPROVED** [review]

**Pull request merged**